src/core/login/create.js

```diff
@@ -92,28 +94,34 @@ export function makeCreateKit(
   const keysKit =
     opts.keyInfo != null ? makeKeysKit(ai, dummyLogin, opts.keyInfo) : {}
 
+  // Secret-key login:
+  const loginAuth = io.random(32)
+  const loginAuthBox = encrypt(io, loginAuth, loginKey)
+  const secretServer: any = wasChangeSecretPayload({
+    loginAuth,
+    loginAuthBox
+  })
+
   // Bundle everything:
   return Promise.all([loginId, passwordKit]).then(values => {
     const [loginIdRaw, passwordKit] = values
     const loginId = base64.stringify(loginIdRaw)
-    const passwordServer: LoginRequestPayload | void = passwordKit.server
-    const pin2Server: LoginRequestPayload | void = pin2Kit.server
-    const keysServer: LoginRequestPayload | void = keysKit.server
-
-    const server: CreateLoginPayload = {
-      appId,
-      loginAuth,
-      loginAuthBox,
-      loginId,
-      parentBox,
-      ...passwordServer,
-      ...pin2Server,
-      ...keysServer
-    }
+    const passwordServer: any = passwordKit.server
+    const pin2Server: any = pin2Kit.server
+    const keysServer: any = keysKit.server
+
     return {
       loginId,
       serverPath: '/v2/login/create',
-      server,
+      server: wasCreateLoginPayload({
+        appId,
+        loginId,
+        parentBox,
+        ...keysServer,
+        ...passwordServer,
+        ...pin2Server,
+        ...secretServer
+      }),
```

Comment:
Move the splats outside the uncleaner `wasCreateLoginPayload` for clarity

Reply:
I'm going to retract the need for this change. On second thought, it seems to make sense that we keep it in the uncleaner input. The reason for this is that we want to allow the extra fields to pass the static type check for wasCreateLoginPayload.